teamup-server/src/main/java/com/project/teamup/service/validator/TechnologyAreaValidator.java

```diff
@@ -0,0 +1,18 @@
+package com.project.teamup.service.validator;
+
+import com.project.teamup.model.TechnologyArea;
+import org.springframework.stereotype.Component;
+
+/**
+ * Validator class for {@link TechnologyArea}.
+ * @author Sebastian
+ */
+@Component
+public class TechnologyAreaValidator {
+
+    public boolean validateObject(TechnologyArea technologyArea){
+        if (technologyArea == null) return false;
+        if (technologyArea.getName().isEmpty()) return false;
+        return technologyArea.getName().length() <= 50;
```

Comment:
As we discussed in the previous meeting for me the annotation validations on the model are more intuitive. If you want to keep the logic in a separate component please extract the values used for fields validaions in constants.